remove_association.hpp

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <phosphor-logging/log.hpp>
+#include <sdbusplus/bus.hpp>
+#include <sdbusplus/bus/match.hpp>
+
+#include <map>
+#include <string>
+#include <variant>
+#include <vector>
+
+namespace phosphor
+{
+namespace inventory
+{
+namespace manager
+{
+
+using namespace phosphor::logging;
+using DBusValue = std::variant<
+    std::string, bool, std::vector<uint8_t>, std::vector<std::string>,
+    std::vector<std::tuple<std::string, std::string, std::string>>>;
+using DBusInterface = std::string;
+using DBusService = std::string;
+using DBusInterfaceList = std::vector<DBusInterface>;
+using DBusObjectPath = std::string;
+using DBusSubtree =
+    std::map<DBusObjectPath, std::map<DBusService, std::vector<DBusInterface>>>;
+
+/**
+ * @class RemoveAssociationBase
+ *
+ * A base class for removing the association set by PEL's
+ *
+ */
+class RemoveAssociationBase
+{
+  public:
+    RemoveAssociationBase() = default;
+    virtual ~RemoveAssociationBase() = default;
+    RemoveAssociationBase(const RemoveAssociationBase&) = default;
+    RemoveAssociationBase& operator=(const RemoveAssociationBase&) = default;
+    RemoveAssociationBase(RemoveAssociationBase&&) = default;
+    RemoveAssociationBase& operator=(RemoveAssociationBase&&) = default;
+
+    /**
+     * @brief Get subtree from the object mapper
+     *
+     * @param[in] bus - The sdbusplus bus object for making D-Bus calls
+     * @param[in] path - The root of the tree to search.
+     * @param[in] depth - The number of path elements to descend.
+     */
+    virtual DBusSubtree listInventoryAssociations(sdbusplus::bus::bus& bus,
+                                                  const std::string& path,
+                                                  int32_t depth) const = 0;
+
+    /**
+     * @brief Removes the critical association on the D-Bus object.
+     *
+     * @param[in] objectPath - The D-Bus object path
+     * @param[in] service - The D-Bus service to call it on
+     */
+    virtual void
+        removeCriticalAssociation(const std::string& objectPath,
+                                  const std::string& service) const = 0;
+};
+
+/**
+ * @class RemoveAssociation
+ *
+ * Concrete implementation of RemoveAssociationBase.
+ */
+class RemoveAssociation : public RemoveAssociationBase
```

Comment:
I don't really see why this has to be a derived class?  In fact, I'm not sure why you need a class at all?  It doesn't store any data.  Really all you need are the plain functions.

Reply:
Done. Removed the class implementation